app/src/main/java/com/ghomebyrw/gworker/fragments/AccountFragment.java

```diff
@@ -72,7 +75,21 @@ public void onDestroyView() {
     void editProfile() {
         Intent intent = new Intent(getContext(), EditProfileActivity.class);
         intent.putExtra("fieldWorker", fieldWorker);
-        startActivity(intent);
+        startActivityForResult(intent, EDIT_PROFILE_REQUEST_CODE);
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == EDIT_PROFILE_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
+            //TODO: call api to update field worker on back end
+            fieldWorker = (FieldWorker) data.getParcelableExtra("fieldWorker");
+            bindFieldWorker(fieldWorker);
+
+            // Force layout refresh. Without this, the right alignment of field values
+            // isn't properly refreshed if the new value is shorter than the old one.
+            // Is there a better way to handle this?
+            layout.invalidate();
```

Comment:
I am not quite sure either, for price update part I started a new activity so the page was refreshed, for the same page I guess it's challenging. 

@codepathreview  @nidhi1608 any suggestions here?  Thanks a lot!